### PR TITLE
Add additional telemetry report counters

### PR DESCRIPTION
**Pull request by duderino**
**What this PR does / why we need it**:

Adds counters to the mixer telemetry reports so we can see how many succeed vs. fail and get a coarse grained understanding of the cause of the failures.

**Which issue this PR fixes**

Helps diagnose issues related to https://github.com/istio/istio/issues/8224

### CODE REVIEW

File: src/istio/mixerclient/report_batch.cc [hunk 1]
```diff
@@ -71,12 +72,34 @@ void ReportBatch::FlushWithLock() {
 
   ++total_remote_report_calls_;
   auto request = batch_compressor_->Finish();
-  ReportResponse* response = new ReportResponse;
+  std::shared_ptr<ReportResponse> response{new ReportResponse()};
```
Comment (duderino) [inline src/istio/mixerclient/report_batch.cc]: note to reviewer, i slipped this in while I was at it.  let me know if you want me to revert

Comment (JimmyCYJ) [reply src/istio/mixerclient/report_batch.cc]: I am fine with this change. Thanks!

**Comment by JimmyCYJ** [comment]
Could you fix the MixerFaultTest.CancelCheck that is failed at macos here or in  a followup PR?

**Comment by duderino** [comment]
@JimmyCYJ yes the PR is waiting for a lgtm:  https://github.com/istio/proxy/pull/2126

**Pull request merged**